From 86aa2578d8a1b8a8af8bb4589a28160c1b1b9fa6 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Sat, 14 Jan 2017 17:15:50 +0100
Subject: [PATCH 1/4] Deduplicate code: Use p_open in p_creat

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 src/win32/posix_w32.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/src/win32/posix_w32.c b/src/win32/posix_w32.c
index fea634b00..8c1aa7840 100644
--- a/src/win32/posix_w32.c
+++ b/src/win32/posix_w32.c
@@ -301,14 +301,7 @@ int p_open(const char *path, int flags, ...)
 
 int p_creat(const char *path, mode_t mode)
 {
-	git_win32_path buf;
-
-	if (git_win32_path_from_utf8(buf, path) < 0)
-		return -1;
-
-	return _wopen(buf,
-		_O_WRONLY | _O_CREAT | _O_TRUNC | STANDARD_OPEN_FLAGS,
-		mode & WIN32_MODE_MASK);
+	return p_open(path, _O_WRONLY | _O_CREAT | _O_TRUNC, mode);
 }
 
 int p_getcwd(char *buffer_out, size_t size)
-- 
2.11.0.windows.1

