From dadbb33b600969319fdc80898d0e4c9ebb16964d Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Fri, 1 Nov 2019 18:55:54 +0100
Subject: [PATCH] Fix crash if snapshotting a config_snapshot

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 src/config_snapshot.c   |  1 +
 tests/config/snapshot.c | 20 ++++++++++++++++++++
 2 files changed, 21 insertions(+)

diff --git a/src/config_snapshot.c b/src/config_snapshot.c
index 764abe208..e89a13886 100644
--- a/src/config_snapshot.c
+++ b/src/config_snapshot.c
@@ -192,6 +192,7 @@ int git_config_backend_snapshot(git_config_backend **out, git_config_backend *so
 	backend->parent.get = config_get_readonly;
 	backend->parent.set = config_set_readonly;
 	backend->parent.set_multivar = config_set_multivar_readonly;
+	backend->parent.snapshot = git_config_backend_snapshot;
 	backend->parent.del = config_delete_readonly;
 	backend->parent.del_multivar = config_delete_multivar_readonly;
 	backend->parent.iterator = config_iterator_new_readonly;
diff --git a/tests/config/snapshot.c b/tests/config/snapshot.c
index 61562d206..3b90cfe49 100644
--- a/tests/config/snapshot.c
+++ b/tests/config/snapshot.c
@@ -100,3 +100,23 @@ void test_config_snapshot__includes(void)
 	cl_git_pass(p_unlink("including"));
 	cl_git_pass(p_unlink("included"));
 }
+
+void test_config_snapshot__snapshot(void)
+{
+	git_config *snapshot_snapshot;
+	int i;
+
+	cl_git_mkfile("configfile", "[section]\nkey = 1\n");
+
+	cl_git_pass(git_config_open_ondisk(&cfg, "configfile"));
+	cl_git_pass(git_config_snapshot(&snapshot, cfg));
+
+	cl_git_pass(git_config_snapshot(&snapshot_snapshot, snapshot));
+
+	cl_git_pass(git_config_get_int32(&i, snapshot_snapshot, "section.key"));
+	cl_assert_equal_i(i, 1);
+
+	git_config_free(snapshot_snapshot);
+
+	cl_git_pass(p_unlink("configfile"));
+}
-- 
2.23.0.windows.1

