From 489591b883f404052dd187962e334342cbebf5d7 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Fri, 23 Jan 2015 17:15:16 +0100
Subject: [PATCH 4/7] Make files with #CR!=#CRLF not fail with
 core.safecrlf=true

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 src/crlf.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/crlf.c b/src/crlf.c
index d88ad00f3..f476fe32a 100644
--- a/src/crlf.c
+++ b/src/crlf.c
@@ -142,12 +142,20 @@ static int crlf_apply_to_odb(
 		if (git_buf_text_gather_stats(&stats, from, false))
 			return GIT_PASSTHROUGH;
 
+		/*
+		 * We're currently not going to even try to convert stuff
+		 * that has bare CR characters. Does anybody do that crazy
+		 * stuff?
+		 */
+		if (stats.cr != stats.crlf)
+			return GIT_PASSTHROUGH;
+
 		/* If there are no CR characters to filter out and CrLf is not set to "true", then just pass */
 		if (!stats.cr && ca->auto_crlf != GIT_AUTO_CRLF_TRUE)
 			return GIT_PASSTHROUGH;
 
 		/* If safecrlf is enabled, sanity-check the result. */
-		if (stats.cr != stats.crlf || stats.lf != stats.crlf) {
+		if (stats.lf != stats.crlf) {
 			switch (ca->safe_crlf) {
 			case GIT_SAFE_CRLF_FAIL:
 				giterr_set(
@@ -175,14 +183,6 @@ static int crlf_apply_to_odb(
 			}
 		}
 
-		/*
-		 * We're currently not going to even try to convert stuff
-		 * that has bare CR characters. Does anybody do that crazy
-		 * stuff?
-		 */
-		if (stats.cr != stats.crlf)
-			return GIT_PASSTHROUGH;
-
 		if (ca->crlf_action == GIT_CRLF_GUESS) {
 			/*
 			 * If the file in the index has any CR in it, do not convert.
-- 
2.16.1.windows.4

