From bb65da7d5cd9fe7d51776dac34af7bbf3a59b7c4 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Sun, 14 Aug 2022 21:37:49 +0200
Subject: [PATCH] Fix memory leak

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 src/util/regexp.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/util/regexp.c b/src/util/regexp.c
index 2569dea0a..08700882b 100644
--- a/src/util/regexp.c
+++ b/src/util/regexp.c
@@ -108,11 +108,11 @@ int git_regexp_match(const git_regexp *r, const char *string)
 	data = pcre2_match_data_create(1, NULL);
 	GIT_ERROR_CHECK_ALLOC(data);
 
-	if ((error = pcre2_match(*r, (const unsigned char *) string, strlen(string),
-			 0, 0, data, NULL)) < 0)
+	error = pcre2_match(*r, (const unsigned char *) string, strlen(string), 0, 0, data, NULL);
+	pcre2_match_data_free(data);
+	if (error < 0)
 		return (error == PCRE2_ERROR_NOMATCH) ? GIT_ENOTFOUND : GIT_EINVALIDSPEC;
 
-	pcre2_match_data_free(data);
 	return 0;
 }
 
-- 
2.37.2.windows.2

