From db3d45c7a5b8773f6221516c015add695708e0fe Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Sun, 29 Jan 2017 02:12:07 +0100
Subject: [PATCH] Update workdir_to_infex CrLf conversion to match Git >=
 2.9 logic

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 src/crlf.c   | 19 +++++++++++--------
 1 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/src/crlf.c b/src/crlf.c
index 468f33b4c..97b29af5c 100644
--- a/src/crlf.c
+++ b/src/crlf.c
@@ -161,9 +161,10 @@ static int crlf_apply_to_odb(
 
 	/* If safecrlf is enabled, sanity-check the result. */
 	if (ca->crlf_action == GIT_CRLF_INPUT ||
-		(ca->auto_crlf == GIT_AUTO_CRLF_INPUT &&
-		(ca->crlf_action == GIT_CRLF_GUESS || ca->crlf_action == GIT_CRLF_AUTO ||
-		(ca->crlf_action == GIT_CRLF_TEXT && ca->eol==GIT_EOL_UNSET)))) {
+		(ca->crlf_action == GIT_CRLF_GUESS || ca->crlf_action == GIT_CRLF_AUTO || ca->crlf_action == GIT_CRLF_TEXT) && ca->eol == GIT_EOL_LF ||
+		(ca->auto_crlf == GIT_AUTO_CRLF_INPUT && (ca->crlf_action == GIT_CRLF_GUESS)) ||
+		(ca->auto_crlf == GIT_AUTO_CRLF_INPUT && (ca->crlf_action == GIT_CRLF_AUTO || ca->crlf_action == GIT_CRLF_TEXT) && ca->eol == GIT_EOL_UNSET) ||
+		(GIT_EOL_NATIVE != GIT_EOL_CRLF && ca->auto_crlf == GIT_AUTO_CRLF_FALSE && (ca->crlf_action == GIT_CRLF_AUTO || ca->crlf_action == GIT_CRLF_TEXT) && ca->eol == GIT_EOL_UNSET)) {
 		if (stats.crlf) {
 			switch (ca->safe_crlf) {
 			case GIT_SAFE_CRLF_FAIL:
@@ -178,9 +179,11 @@ static int crlf_apply_to_odb(
 				break;
 			}
 		}
-	} else if (ca->crlf_action == GIT_CRLF_CRLF ||
-				(ca->auto_crlf == GIT_AUTO_CRLF_TRUE && ca->crlf_action == GIT_CRLF_GUESS ||
-				((ca->crlf_action == GIT_CRLF_TEXT || ca->crlf_action == GIT_CRLF_AUTO) && ca->eol == GIT_EOL_UNSET))) {
+	} else if ((ca->crlf_action == GIT_CRLF_CRLF ||
+		((ca->crlf_action == GIT_CRLF_GUESS || ca->crlf_action == GIT_CRLF_AUTO || ca->crlf_action == GIT_CRLF_TEXT) && ca->eol == GIT_EOL_CRLF) ||
+		(ca->auto_crlf == GIT_AUTO_CRLF_TRUE && (ca->crlf_action == GIT_CRLF_GUESS)) ||
+		(ca->crlf_action == GIT_CRLF_AUTO || ca->crlf_action == GIT_CRLF_TEXT)) &&
+		!(GIT_EOL_NATIVE != GIT_EOL_CRLF && ca->auto_crlf == GIT_AUTO_CRLF_FALSE && (ca->crlf_action == GIT_CRLF_AUTO || ca->crlf_action == GIT_CRLF_TEXT) && ca->eol == GIT_EOL_UNSET)) {
 		if (stats.lf != stats.crlf) {
 			switch (ca->safe_crlf) {
 			case GIT_SAFE_CRLF_FAIL:
@@ -321,8 +324,6 @@ static int crlf_check(
 	 * Use the core Git logic to see if we should perform CRLF for this file
 	 * based on its attributes & the value of `core.autocrlf`
 	 */
-	ca.crlf_action = crlf_input_action(&ca);
-
 	if (ca.crlf_action == GIT_CRLF_BINARY)
 		return GIT_PASSTHROUGH;
 
@@ -330,6 +331,8 @@ static int crlf_check(
 		((ca.crlf_action == GIT_CRLF_AUTO || ca.crlf_action == GIT_CRLF_TEXT) &&
 		git_filter_source_mode(src) == GIT_FILTER_SMUDGE)) {
 
+		ca.crlf_action = crlf_input_action(&ca);
+
 		if (ca.crlf_action == GIT_CRLF_GUESS &&
 			ca.auto_crlf == GIT_AUTO_CRLF_FALSE)
 			return GIT_PASSTHROUGH;
-- 
2.11.0.windows.3

