From 3a766cdf7533a9a258fafee1c145be2035a36441 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Sat, 28 Jan 2017 18:31:11 +0100
Subject: [PATCH] Update index_to_workdir CrLf conversion to match Git >= 2.9
 logic

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 src/crlf.c   | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/crlf.c b/src/crlf.c
index b8ae5cda1..9c3db570a 100644
--- a/src/crlf.c
+++ b/src/crlf.c
@@ -70,6 +70,9 @@ static int crlf_input_action(struct crlf_attrs *ca)
 	if (ca->eol == GIT_EOL_LF)
 		return GIT_CRLF_INPUT;
 
+	if (ca->crlf_action == GIT_CRLF_AUTO)
+		return GIT_CRLF_AUTO;
+
 	if (ca->eol == GIT_EOL_CRLF)
 		return GIT_CRLF_CRLF;
 
@@ -199,6 +202,8 @@ static const char *line_ending(struct crlf_attrs *ca)
 		break;
 
 	case GIT_CRLF_AUTO:
+		if (ca->eol == GIT_EOL_CRLF)
+			return "\r\n";
 	case GIT_CRLF_TEXT:
 		break;
 
@@ -252,8 +257,7 @@ static int crlf_apply_to_workdir(
 		ca->crlf_action == GIT_CRLF_GUESS) {
 
 		/* If we have any existing CR or CRLF line endings, do nothing */
-		if (ca->crlf_action == GIT_CRLF_GUESS &&
-			stats.cr > 0 && stats.crlf > 0)
+		if (stats.cr > 0 && stats.crlf > 0)
 			return GIT_PASSTHROUGH;
 
 		/* If we have bare CR characters, do nothing */
@@ -313,7 +317,7 @@ static int crlf_check(
 			ca.auto_crlf == GIT_AUTO_CRLF_FALSE)
 			return GIT_PASSTHROUGH;
 
-		if (ca.auto_crlf == GIT_AUTO_CRLF_INPUT &&
+		if (ca.auto_crlf == GIT_AUTO_CRLF_INPUT && ca.eol != GIT_EOL_CRLF &&
 			git_filter_source_mode(src) == GIT_FILTER_SMUDGE)
 			return GIT_PASSTHROUGH;
 	}
-- 
2.11.0.windows.3

