From 397bc237c63758e6818e4fc08c7a97bb39270419 Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Sat, 20 Jun 2015 17:00:03 +0200
Subject: [PATCH 7/7] Allow to use %PROGRAMDATA%\Git\config file

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 src/config.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/config.c b/src/config.c
index 77cf573..12ec5a4 100644
--- a/src/config.c
+++ b/src/config.c
@@ -1082,6 +1082,9 @@ int git_config_find_xdg(git_buf *path)
 int git_config_find_system(git_buf *path)
 {
 	git_buf_sanitize(path);
+	/* system config in %PROGRAMDATA%\Git on win32 is named config, see https://github.com/git-for-windows/git/commit/ba8c0399e60807f08f73ce6a394c72c261eb96ce */
+	if (!git_sysdir_find_system_file(path, "config"))
+		 return 0;
 	return git_sysdir_find_system_file(path, GIT_CONFIG_FILENAME_SYSTEM);
 }
 
-- 
1.8.1.2

