From 5b7e71314b6e232d117aa7c07b1f91f9ae063a6a Mon Sep 17 00:00:00 2001
From: Eric Huss <eric@huss.org>
Date: Wed, 7 Oct 2020 15:54:08 -0700
Subject: [PATCH] Fix error return for invalid extensions.

---
 src/repository.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/repository.c b/src/repository.c
index a1186832b..513dbd61f 100644
--- a/src/repository.c
+++ b/src/repository.c
@@ -860,7 +860,7 @@ int git_repository_open_ext(
 	if (config && (error = check_repositoryformatversion(&version, config)) < 0)
 		goto cleanup;
 
-	if ((error = check_extensions(config, version) < 0))
+	if ((error = check_extensions(config, version)) < 0)
 		goto cleanup;
 
 	if ((flags & GIT_REPOSITORY_OPEN_BARE) != 0)
@@ -1613,7 +1613,7 @@ static int repo_init_config(
 	if (is_reinit && (error = check_repositoryformatversion(&version, config)) < 0)
 		goto cleanup;
 
-	if ((error = check_extensions(config, version) < 0))
+	if ((error = check_extensions(config, version)) < 0)
 		goto cleanup;
 
 #define SET_REPO_CONFIG(TYPE, NAME, VAL) do { \
-- 
2.28.0.windows.1

