From a7e6c4d7d81c3606a051aabac49768a52b4c7dce Mon Sep 17 00:00:00 2001
From: Sven Strickroth <email@cs-ware.de>
Date: Sat, 9 Dec 2023 16:32:55 +0100
Subject: [PATCH] Fix warning "'ENABLE_INTSAFE_SIGNED_FUNCTIONS': macro
 redefinition"

Signed-off-by: Sven Strickroth <email@cs-ware.de>
---
 src/util/integer.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/util/integer.h b/src/util/integer.h
index 63277177b..a9e416cc3 100644
--- a/src/util/integer.h
+++ b/src/util/integer.h
@@ -89,7 +89,9 @@ GIT_INLINE(int) git__is_int(int64_t p)
 /* Use Microsoft's safe integer handling functions where available */
 #elif defined(_MSC_VER)
 
-# define ENABLE_INTSAFE_SIGNED_FUNCTIONS
+# if !defined(ENABLE_INTSAFE_SIGNED_FUNCTIONS)
+#  define ENABLE_INTSAFE_SIGNED_FUNCTIONS
+# endif
 # include <intsafe.h>
 
 # define git__add_sizet_overflow(out, one, two) \
-- 
2.43.0.windows.1

